### ISSUE TITLE
Unable to Reproduce results using ILORA.py

### ISSUE THREAD

**Issue by varunlmxd** [which47/LLMCL#10]
There are so many issues to use ILORA setup 
1. Unable to use load_best_model_at_end because after every epoch every dataset is eval loss is calculated.
2. Unable to load the saved model after 1 epoch, In ILORA.py line no 300 
```
    def save_model(self, name) -> str:
        if self.args.output_dir is not None:
    Here -> output_dir = os.path.join(self.args.output_dir, f"{self.cl_method}_{self.adapter}_checkpoint_{name}")
            self.model.model.set_adapter('default')
            self.model.model.save_pretrained(output_dir)
            return output_dir
```
3. Error at ILORA.py line no 295
```
    def continual_learning(self):
        resume_from_checkpoint = "False"
        for task_name, dataset in self.continual_training_dataset.items():
            self.model.current_task_name = task_name
            self.current_task_name = task_name
            self.train_dataset = dataset
            self.train()
            resume_from_checkpoint = self.save_model(task_name)
   Here - > self.model.load_ema_model(resume_from_checkpoint)
        wandb.finish()

```

there is no load_ema_model should it be replaced with update_ema_model? Beacause I have replaced with update_ema_model.
